packages/components/modules/notifications/NotificationsList/index.tsx

```diff
@@ -0,0 +1,130 @@
+
+  const options = { count: 10 }
+  const { me } = useLazyLoadQuery<NotificationsListQueryType>(NotificationsListQuery, options, {
+    fetchPolicy: 'store-and-network',
+  })
+
+  const { data, loadNext, isLoadingNext, hasNext, refetch } = usePaginationFragment<
+    NotificationsListQueryType,
+    NotificationsListFragment$key
+  >(NotificationsListFragment, me)
+
+  useNotificationsSubscription(me?.id)
+
+  const notifications = useMemo(
+    () => data?.notifications?.edges.filter((edge) => edge?.node).map((edge) => edge?.node) || [],
+    [data?.notifications?.edges],
+  )
+  const renderNotificationItem = (notification: any, index: number) => {
+    if (!notification) return null
+    if (!notification.unread && notifications[index - 1]?.unread) {
+      return (
+        <>
+          <Divider>
+            <Typography variant="body2">Older</Typography>
+          </Divider>
+          <NotificationItem notification={notification} key={`notification-${notification.id}`} />
+        </>
+      )
+    }
+    return <NotificationItem notification={notification} key={`notification-${notification.id}`} />
+  }
+
+  const renderVirtuosoHeader = () => <div className="h-2" />
+
+  const renderVirtuosoLoadingState = () => {
+    if (!isLoadingNext) return <Box sx={{ paddingTop: 3 }} />
+
+    return (
+      <LoadingState
+        sx={{ paddingTop: 3 }}
+        CircularProgressProps={{ size: 15 }}
+        {...LoadingStateProps}
+      />
+    )
+  }
+
+  const renderContent = () => {
+    if (notifications.length === 0) return <EmptyState />
+
+    return (
+      <Box sx={{ backgroundColor: 'common.white' }}>
+        <Virtuoso
+          data={notifications}
+          overscan={NUMBER_OF_NOTIFICATIONS_TO_LOAD_NEXT}
+          style={{ height: 'calc(100vh - 68px)' }}
+          itemContent={(index, notification) => renderNotificationItem(notification, index)}
+          components={{
+            Header: renderVirtuosoHeader,
+            Footer: renderVirtuosoLoadingState,
+          }}
+          endReached={() => {
+            if (hasNext) {
+              loadNext(NUMBER_OF_NOTIFICATIONS_TO_LOAD_NEXT)
+            }
+          }}
+        />
+      </Box>
+    )
+  }
+
+  const refetchNotifications = () => {
+    refetch(options, { fetchPolicy: 'network-only' })
+  }
+
+  return (
+    <Box display="grid" gridTemplateRows="min-content 1fr" height="100%">
+      <HeaderContainer>
+        <Typography variant="h6">Notifications</Typography>
+        <Box display="grid" gridTemplateColumns="max-content min-content">
+          {!!data?.notificationsUnreadCount && (
+            <MarkAllAsReadButton refetch={refetchNotifications} />
+          )}
+          {smDown && (
+            <IconButton onClick={() => setIsDrawerOpened(false)}>
+              <CloseIcon />
+            </IconButton>
+          )}
+        </Box>
+      </HeaderContainer>
+      {renderContent()}
+    </Box>
+  )
+}
+
+export default NotificationsList
```

Comment:
we need to wrap this component in a Suspense boundary so we can show the proper loading (instead of triggering the application suspense boundary, which in the case of baseapp-template is `app/loading.tsx`)

https://relay.dev/docs/api-reference/use-lazy-load-query/
https://relay.dev/docs/guided-tour/rendering/loading-states/

```
const NotificationsListSuspended: FC<NotificationsListProps> = (props) => (
  <Suspense fallback={<LoadingState />}>
    <NotificationsList {...props} />
  </Suspense>
)

export default NotificationsListSuspended
```

Reply:
nice catch, I updated